includes/class-amp-story-post-type.php

```diff
@@ -174,2197 +102,170 @@ public static function register() {
-						$style_attr_values[ $hash ] = self::safecss_filter_attr( wp_unslash( $matches['styles'] ) );
-
-						// Replaces the complete style attribute value with its hashed version.
-						return $matches['before'] . sprintf( ' data-temp-style-hash="%s" ', $hash ) . $matches['after'];
-					},
-					$post_content
-				);
-
-				return $post_content;
-			},
-			0
-		);
+		$asset_file   = AMP__DIR__ . '/assets/js/' . self::AMP_STORIES_SCRIPT_HANDLE . '.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
 
-		// ...And bring it back afterwards.
-		add_filter(
-			'content_save_pre',
-			static function ( $post_content ) use ( &$style_attr_values ) {
-				// Replaces hashed style attribute value with the original value again.
-				return preg_replace_callback(
-					'/ data-temp-style-hash=\\\"(?P<hash>[0-9a-f]+)\\\"/',
-					static function ( $matches ) use ( $style_attr_values ) {
-						return isset( $style_attr_values[ $matches['hash'] ] ) ? sprintf( ' style="%s"', esc_attr( wp_slash( $style_attr_values[ $matches['hash'] ] ) ) ) : '';
-					},
-					$post_content
-				);
-			},
-			20
+		wp_enqueue_script(
+			self::AMP_STORIES_SCRIPT_HANDLE,
+			amp_get_asset_url( 'js/' . self::AMP_STORIES_SCRIPT_HANDLE . '.js' ),
+			$dependencies,
+			$version,
+			false
 		);
 
-		return $response;
-	}
-
-	/**
-	 * Filter the allowed tags for KSES to allow for amp-story children.
-	 *
-	 * @param array $allowed_tags Allowed tags.
-	 * @return array Allowed tags.
-	 */
-	public static function filter_kses_allowed_html( $allowed_tags ) {
-		$story_components = [
-			'amp-story-page',
-			'amp-story-grid-layer',
-			'amp-story-cta-layer',
-			'amp-img',
-			'amp-video',
-			'img',
-		];
-		foreach ( $story_components as $story_component ) {
-			$attributes = array_fill_keys( array_keys( AMP_Allowed_Tags_Generated::get_allowed_attributes() ), true );
-			$rule_specs = AMP_Allowed_Tags_Generated::get_allowed_tag( $story_component );
-			foreach ( $rule_specs as $rule_spec ) {
-				$attributes = array_merge( $attributes, array_fill_keys( array_keys( $rule_spec[ AMP_Rule_Spec::ATTR_SPEC_LIST ] ), true ) );
-			}
-			$allowed_tags[ $story_component ] = $attributes;
-		}
-
-		// @todo This perhaps should not be allowed if user does not have capability.
-		foreach ( $allowed_tags as &$allowed_tag ) {
-			$allowed_tag['animate-in']           = true;
-			$allowed_tag['animate-in-duration']  = true;
-			$allowed_tag['animate-in-delay']     = true;
-			$allowed_tag['animate-in-after']     = true;
-			$allowed_tag['data-font-family']     = true;
-			$allowed_tag['data-block-name']      = true;
-			$allowed_tag['data-temp-style-hash'] = true;
-			$allowed_tag['layout']               = true;
-			$allowed_tag['object-position']      = true;
+		if ( function_exists( 'wp_set_script_translations' ) ) {
+			wp_set_script_translations( 'amp-editor-story-blocks-build', 'amp' );
```

Comment:
@swissspidy Do we need to change the `amp-editor-story-blocks-build` handle here? I can't trace how this translations file down.

Reply:
> wp_set_script_translations is not locked the block. This is just any translations found in a script.

I know, that was a successor to the legacy Gutenberg plugin functions which were removed.

What I'm saying is that we're not using the `amp-editor-story-blocks-build` anymore here. That was something we added ages ago for the Gutenberg blocks used in the Stories editor specifically.

Reply:
Currently, only `amp-edit-story` is used, if we'll need to add more scripts in the future then we can add it later, keeping this PR as small as necessary at this moment.

Reply:
So just remove the whole thing?

Reply:
Yep!

Reply:
Done - https://github.com/ampproject/amp-wp/pull/3839/commits/6519a9653cc8798ef435a88ceac83219fcd8e48b

Reply:
Looks like you removed it from the legacy post type and not from the new :)

Reply:
Restored https://github.com/ampproject/amp-wp/pull/3839/commits/1b69f11cb7d2e03c385db69bcea1b434d83bbbac

:man_facepalming: